CMakeLists.txt

```diff
@@ -44,7 +44,18 @@ set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
 
 ### Defaults
 
-set(ACADOS_INSTALL_DIR "${PROJECT_SOURCE_DIR}" CACHE PATH  "Installation path to PROJECT_SOURCE_DIR")
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  if(DEFINED ACADOS_INSTALL_DIR)
+    message(WARNING "ACADOS_INSTALL_DIR variable is deprecated, use CMAKE_INSTALL_PREFIX instead.")
+    set_property(CACHE CMAKE_INSTALL_PREFIX PROPERTY VALUE "${ACADOS_INSTALL_DIR}")
+  else()
+    set_property(CACHE CMAKE_INSTALL_PREFIX PROPERTY VALUE "${PROJECT_SOURCE_DIR}")
+  endif()
+else()
+  if(DEFINED ACADOS_INSTALL_DIR)
+    message(WARNING "Deprecated ACADOS_INSTALL_DIR variable has been overriden by CMAKE_INSTALL_PREFIX: ${CMAKE_INSTALL_PREFIX}")
```

Comment:
Can we make `ACADOS_INSTALL_DIR` have priority over `CMAKE_INSTALL_PREFIX` in a first step?
If I pull your branch and do my usual installation, I get:

```
CMake Warning at CMakeLists.txt:56 (message):
  Deprecated ACADOS_INSTALL_DIR variable has been overriden by
  CMAKE_INSTALL_PREFIX: /usr/local
```
This will result in an error for all users that use the default installation process.

I would rather require people who use `CMAKE_INSTALL_PREFIX` directly to clean their cmake cache, since this should be less acados users.
Does this make sense?

Also, there is a typo in `overriden`.

Reply:
ok, new version should behave the way you want.